### DISCUSSION TITLE
Connection Timeout beim Start (via Docker)

### DISCUSSION THREAD

**Discussion by christiangartner** [Zefau/ioBroker.jarvis#1729] [Q&A]
Hallo, 

ich habe iobroker über Docker auf einer Synology laufen. Leider kommt beim Start von jarvis immer ein Connection Timeout. 
Error 501 - Not implemented

<img width="640" alt="Screenshot 2022-06-18 at 12 10 33" src="https://user-images.githubusercontent.com/20615556/174433315-01b160cc-d185-48c9-b3b9-6f04ec964601.png">

Meine Portsettings:
<img width="300" alt="Screenshot 2022-06-18 at 12 16 50" src="https://user-images.githubusercontent.com/20615556/174433439-99dc7b5a-ce09-4d38-9cb2-17d59146c312.png">

Jarvis Adapter:
<img width="400" alt="Screenshot 2022-06-18 at 12 19 03" src="https://user-images.githubusercontent.com/20615556/174433490-da4b7946-3729-4c73-a141-d4cfc54da73f.png">

Web:
<img width="550" alt="Screenshot 2022-06-18 at 12 20 53" src="https://user-images.githubusercontent.com/20615556/174433557-11d2b7d9-715f-43d2-81e8-e7d6595e6d16.png">

Socket.io:
<img width="400" alt="Screenshot 2022-06-18 at 12 21 06" src="https://user-images.githubusercontent.com/20615556/174433567-0e65952e-8de4-46f9-ae4c-1212806c912d.png">

Komme selbst nicht weiter, ev. kann mir jemand auf die Sprünge helfen.

Lg
Christian

**Comment by Zefau** [comment]
Welche Version hast du im Einsatz? Bei V3 musst du 8400 Mappen, das ist der Socket Port. Der Socket-IO Adapter findet ab V3 keine Anwendung mehr.

**Reply by christiangartner** [reply]
Danke für die schnelle Antwort u. den Hinweis, habe Version 3.0.11 im Einsatz. 

Folgendes versucht:
- Socket.io deaktiviert.
- Docker Port 49156 auf 8400 gemappt
- Jarvis-Adapter config als Socketport 49156 eingetragen.
- Web-Adapter Socket.io Instanz auf "none" gestellt.
->
<img width="450" alt="Screenshot 2022-06-18 at 16 23 10" src="https://user-images.githubusercontent.com/20615556/174443562-4e7dc5ae-812a-4ae5-8604-e169fdf6b948.png">

ERR_CONNECTION_RESET u. ERR_EMPTY_RESPONSE

2. Versuch Adapter Socket.io Instanz auf "integrated", selber Output.

**Reply by Zefau** [reply]
Konfiguration sieht richtig aus. Hast du Discord?

Was passiert wenn du die URL zum Socket Port direkt öffnest?

**Comment by Zefau** [comment]
Generell auf ioBroker kommst du rauf?

**Reply by christiangartner** [reply]
Hab den Fehler gefunden: 

1. In der Jarvis-Adapterconfig ist als Socketport 8400 zu setzen (statt den von außen erreichbaren gemappten in Docker (49156))
2. Beim Aufruf des Weblinks vom Adapter ist die URL dann aber manuell anzupassen u. Socketport auf den gemappten Port zu setzen. (Hier muss man schnell sein die URL zu kopieren vor dem redirect.) http://gnas.local:49154/jarvis/index.html#/0/&socketHost=gnas.local&socketPort=8400&socketSecure=false zu 
http://gnas.local:49154/jarvis/index.html#/0/&socketHost=gnas.local&socketPort=49156&socketSecure=false

Meine Interpretation dazu:
- In der Adapterconfig wird nur definiert auf welchen Socketport der Jarvis-Endpoint horcht. -> 8400
- In der URL wird definiert mit welchen Port der Client (Browser) den Server aufruft. -> gemappter Port (in meinem Fall 49156)


Danke für die Unterstützung
